src/vm/createVm.spec.js

```diff
@@ -0,0 +1,227 @@
+/* eslint-env jest */
+
+import {
+  find
+} from 'lodash'
+import eventToPromise from 'event-to-promise'
+
+import {
+  config,
+  getNetworkId,
+  getSrId,
+  waitObjectState,
+  xo
+} from './../util'
+
+// ===================================================================
+
+describe('.create()', () => {
+  let vmId
+  let networkId
+  let srId
+  let serverId
+
+  // ----------------------------------------------------------------------
+
+  beforeEach(() => {
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = 30e3
+  })
+
+  // ----------------------------------------------------------------------
+
+  async function createVm (params) {
+    const vmId = await xo.call('vm.create', params)
+    return vmId
+  }
+
+  async function createVmTest () {
+    const templateId = getTemplateId(config.templates.debian)
+    const vmId = await createVm({
+      name_label: 'vmTestN',
+      template: templateId,
+      VIFs: []
+    })
+    return vmId
+  }
+
+  function deleteVm (id) {
+    return xo.call('vm.delete', {id, delete_disks: true})
+  }
+
+  function getTemplateId (nameTemplate) {
+    const template = find(xo.objects.all, {type: 'VM-template', name_label: nameTemplate})
+    return template.id
+  }
+
+  // =================================================================
+
+  describe('.createVmMaster()', () => {
+    it.only('creates a VM on a master server', async () => {
+      serverId = await xo.call('server.add', config.masterServer).catch(() => {})
+      await eventToPromise(xo.objects, 'finish')
```

Comment:
We need to wait adding/updating objects in our collection. `objects.set(id, object)`.